bindings/go/scip/speedtest/speedtest_main.go

```diff
@@ -0,0 +1,188 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"runtime"
+	"runtime/debug"
+	"time"
+
+	"github.com/jedib0t/go-pretty/v6/table"
+	"github.com/jedib0t/go-pretty/v6/text"
+	"google.golang.org/protobuf/proto"
+
+	"github.com/sourcegraph/beaut"
+	"github.com/sourcegraph/beaut/lib/knownwf"
+
+	"github.com/sourcegraph/scip/bindings/go/scip"
+	"github.com/sourcegraph/scip/bindings/go/scip/internal"
+	"github.com/sourcegraph/scip/bindings/go/scip/internal/shared"
+)
+
+// NOTE: We don't run this benchmark using the standard
+// Go benchmarking tooling because that doesn't offer
+// any control over the number of iterations performed.
+func main() {
+	for _, path := range shared.SampleIndexes() {
+		benchmark(path)
+	}
+}
+
+func benchmark(path string) {
+	scipReader, err := os.Open(path)
+	if err != nil {
+		panic(fmt.Sprintf("failed to open file %q: %v", path, err.Error()))
+	}
+	scipBytes, err := io.ReadAll(scipReader)
+	if err != nil {
+		panic(fmt.Sprintf("failed to read file %q: %v", path, err.Error()))
+	}
+	scipIndex := scip.Index{}
+	if err := proto.Unmarshal(scipBytes, &scipIndex); err != nil {
+		panic(fmt.Sprintf("failed to unmarshal index %q: %v", path, err.Error()))
+	}
+	allOccurrences := []*scip.Occurrence{}
+	for _, document := range scipIndex.Documents {
+		allOccurrences = append(allOccurrences, document.Occurrences...)
+	}
+	parserV1Benchmark := func(b *simpleBenchmark) {
+		for i := 0; i < b.N; i++ {
+			occ := allOccurrences[i]
+			_, err = internal.ParsePartialSymbolV1ToBeDeleted(occ.Symbol, true)
+			if err != nil {
+				//panic(fmt.Sprintf("v1: index path: %v: error: %v", path, err))
```

Comment:
Should we at least collect these errors, and check that both parsers errored on the same symbols?

Reply:
That is handled by TestParseCompat, we're deliberately dropping them here. Added a comment explaining that.